protocols/imap/src/main/java/org/apache/james/imap/processor/AuthenticateProcessor.java

```diff
@@ -155,4 +163,28 @@ protected Closeable addContextToMDC(AuthenticateRequest request) {
             .addToContext("authType", request.getAuthType())
             .build();
     }
+
+    private void doOAuth(String jwtToken, ImapSession session, ImapRequest request, Responder responder) {
+        Preconditions.checkArgument(session.oidcSaslConfiguration().isPresent(), "Can not do SASL when the oidcSasl configuration is not yet");
+        OidcSASLConfiguration configuration = session.oidcSaslConfiguration().get();
+        Optional<String> claim = new OidcJwtTokenVerifier().verifyAndExtractClaim(jwtToken, configuration.getJwksURL(), configuration.getClaim());
+
+        if (claim.isPresent()) {
+            final MailboxManager mailboxManager = getMailboxManager();
+            try {
+                Username username = Username.fromMailAddress(new MailAddress(claim.get()));
+                MailboxSession mailboxSession = mailboxManager.createSystemSession(username);
```

Comment:
Should we use `createSystemSession`? 
systemSession is not the right SessionType for this case. It should be `MailboxSession.SessionType.User` 
Should we have a new method in SessionProvider? (`createSesison(Username userName, MailboxSession.SessionType type)`)

Reply:
Yes that's ok.

Reply:
> Should we have a new method in SessionProvider? (createSesison(Username userName, MailboxSession.SessionType type))

That's costly. Rene had been pushing for something similar a while ago but it ended up being non-consensual...

Reply:
Remove final qualifier usages for variables